consumer.go

```diff
@@ -106,7 +106,8 @@ type Consumer struct {
 	channel string
 	config  Config
 
-	rng *rand.Rand
+	rngMtx sync.RWMutex
```

Comment:
`RWMutex` is designed for separate read and write locks. Since we only want global locking can you use `sync.Mutex` here?